app/grandchallenge/components/views.py

```diff
@@ -330,6 +330,7 @@ class CivSetListView(
 ):
     model = None
     permission_required = None
+    delete_all_enabled = True

```

Comment:
Why do you need to set this as class variable?

Reply:
I saw it as a static configuration (always True) on the base view class level `CivSetListView` so I made as a class variable. Looking at again, and knowing that its override will be dynamic and involves a database query (in `ReaderStudyDisplaySetList`) I think it should be a property (see below). What do you think?

![Screenshot 2024-09-03 at 13 33 58](https://github.com/user-attachments/assets/5695c5ce-5d38-4e90-94cc-595c3b925ea7)

What I am not sure about is, should we check for the delete_all_enabled and add the message to the context only when it will be viewed, or it is okay to add anyway and just do the check in the template?

Reply:
You're making this far too complicated. Most straightforward I can think of is `app/grandchallenge/reader_studies/views.py`:

```python
def get_context_data(self, *args, **kwargs):
    context = super().get_context_data(*args, **kwargs)
    
    if Answer.objects.filter(question__reader_study=self.base_object).exists():
        context["delete_all_disabled_message"] = "Cannot delete all display sets: first you need to delete all of the answers for this reader study"

    return context
```

in `app/grandchallenge/components/templates/components/civ_set_list.html`:

```
{% if delete_all_disabled_message %}
   disabled
   title="{{ delete_all_disabled_message }}"
{% endif %}
```